android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java

```diff
@@ -119,41 +120,53 @@ public void onDismiss() {
         setUpViews();
     }
 
-    public void showPopupMenu() {
-        PopupMenu menu = new PopupMenu(mOptionsImage.getContext(), (View) mOptionsImage);
-        menu.getMenuInflater().inflate(R.menu.menu_dapps_panel, menu.getMenu());
-        menu.setOnMenuItemClickListener(item -> {
-            if (item.getItemId() == R.id.action_lock_wallet) {
-                mBraveWalletPanelServices.getKeyringService().lock();
+    private void showPopupMenu() {
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.N_MR1) {
+            androidx.appcompat.widget.PopupMenu menu = new androidx.appcompat.widget.PopupMenu(
+                    mOptionsImage.getContext(), (View) mOptionsImage);
+            menu.getMenuInflater().inflate(R.menu.menu_dapps_panel, menu.getMenu());
+            menu.setOnMenuItemClickListener(item -> { return handleMenuItemClick(item); });
+
+            if (menu.getMenu() instanceof MenuBuilder) {
+                ((MenuBuilder) menu.getMenu()).setOptionalIconsVisible(true);
+            }
+            menu.show();
+        } else {
+            android.widget.PopupMenu menu = new android.widget.PopupMenu(
+                    mAnchorViewHost.getContext(), (View) mAnchorViewHost);
+            menu.getMenuInflater().inflate(R.menu.menu_dapps_panel, menu.getMenu());
+            menu.setOnMenuItemClickListener(item -> { return handleMenuItemClick(item); });
+            menu.show();
```

Comment:
not sure but might be possible to use internal [Menubuilder](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/com/android/internal/view/menu/MenuBuilder.java;drc=349d6efeed02e20c23093f73b3d1ea913ec43632;l=53) here to show the icon. also not sure if it shows the icon without the menubuilder?

Reply:
yeah, not sure, anyways, I fixed the crash, we can always polish the UI more if we need to, we have only a few % of users on Android 6 and 7, I don't think it's worth spending time on polishing that more. The main functionality is working.